app/models/credentials.rb

```diff
@@ -63,8 +70,7 @@ def validate
 
     validates_presence [ :api_key ]
 
-    errors.add(:password, 'must not be blank') if @plain_password && @plain_password.empty?
-    errors.add(:password, 'cannot contain a newline') if @plain_password && @plain_password.index("\n")
+    errors.add(:password, ::Errors::Conjur::InsufficientPasswordComplexity.new.to_s) if @plain_password && (@plain_password.index("\n") || @plain_password !~ VALID_PASSWORD_REGEX)
```

Comment:
Let's  use the Sequel Gem validation: https://github.com/jeremyevans/sequel/blob/master/doc/validations.rdoc#validates_format
It would be something like:
```
validates_format VALID_PASSWORD_REGEX, : plain_password
```

Reply:
The scope of `validate` method includes both `api_key` and `plain_password`. If we add `validate_format` it will not separate between the two use-cases. 

See existing password validation logic is to verify existence of `plain_password` parameter, and only after, verify complexity, blank & new line.

We also wanted to utilize `validates_presence`, but neglect the idea for the same reason above.